lib/shopify-cli/php_deps.rb

```diff
@@ -0,0 +1,99 @@
+require "shopify_cli"
+
+module ShopifyCli
+  ##
+  # ShopifyCli::PHPDeps ensures that all PHP dependencies are installed for projects.
+  #
+  class PHPDeps
+    include SmartProperties
+
+    property! :ctx, accepts: ShopifyCli::Context
+
+    ##
+    # Proxy to instance method ShopifyCli::PHPDeps.new.install.
+    #
+    # #### Parameters
+    # - `ctx`: running context from your command
+    # - `verbose`: whether to run the installation tools in silent mode
+    #
+    # #### Example
+    #
+    #   ShopifyCli::PHPDeps.install(ctx)
+    #
+    def self.install(ctx, verbose = false)
+      new(ctx: ctx).install(verbose)
+    end
+
+    ##
+    # Installs all of a project's PHP dependencies using Composer.
+    #
+    # #### Parameters
+    # - `verbose`: whether to run the installation tools in silent mode
+    #
+    # #### Example
+    #
+    #   # context is the running context for the command
+    #   ShopifyCli::PHPDeps.new(context).install(true)
+    #
+    def install(verbose = false)
+      title = ctx.message("core.php_deps.installing")
+      success = ctx.message("core.php_deps.installed")
+      failure = ctx.message("core.php_deps.error.install_error")
+
+      CLI::UI::Frame.open(title, success_text: success, failure_text: failure) do
+        composer(verbose)
+      end
+    end
+
+    private
+
+    def composer(verbose = false)
+      cmd = %w(composer install)
+      cmd << "--quiet" unless verbose
+
+      run_install_command(cmd)
+    end
+
+    def run_install_command(cmd)
+      deps = parse_dependencies
+      errors = nil
+
+      spinner_title = ctx.message("core.php_deps.installing")
+      success = CLI::UI::Spinner.spin(spinner_title, auto_debrief: false) do |spinner|
+        _, errors, status = CLI::Kit::System.capture3(*cmd, env: @ctx.env, chdir: ctx.root)
```

Comment:
It would probably be helpful to give the unused property a name anyway so we know what we aren't using